custom_components/thermal_comfort/config_flow.py

```diff
@@ -21,6 +27,28 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+def get_sensors_by_device_class(
+    _er: EntityRegistry,
+    _hass: HomeAssistant,
+    device_class: str,
+    include_not_in_registry: bool = False,
+) -> list:
+    """Get sensors of required class from entity registry."""
+
+    result = []
+    for d in _er.async_get_device_class_lookup(
+        {(Platform.SENSOR, device_class)}
+    ).values():
+        for e in d.values():
+            result.append(e)
+
+    if include_not_in_registry:
+        result += list(
+            {e.entity_id for e in _hass.states.async_all()} - set(_er.entities)
+        )
```

Comment:
doesn't work as intended, to many sensors missing compared to just
```python
        result += [e.entity_id for e in _hass.states.async_all()]
```

Reply:
> doesn't work as intended, to many sensors missing compared to just

Yes, many sensor missed. And it is main idea: all sensors from registry have device class, so this add only out of device class sensors. It allow us to skip pressure sensors from list of temperature, for example.

Reply:
Yes but there are also temperature sensors completely missing that make no sense not to be available in the list.
e.g. this one
![image](https://user-images.githubusercontent.com/1131371/151427371-b49d274a-0204-4e5c-940a-175d2c247d75.png)

Reply:
> Yes but there are also temperature sensors completely missing that make no sense not to be available in the list. e.g. this one ![image](https://user-images.githubusercontent.com/1131371/151427371-b49d274a-0204-4e5c-940a-175d2c247d75.png)

Is this sensor in device registry, but have no device_class `DEVICE_CLASS_TEMPERATURE`?

Reply:
You can get that one with:
```yaml
sensor:
  - platform: demo
```

Reply:
Will test this setup.

Reply:
```json
{
   "entity_id=""sensor.outside_temperature",
   "unique_id=""sensor_1",
   "platform=""demo",
   "area_id=""living_room",
   "capabilities="{
      "state_class":"measurement"
   },
   "config_entry_id=None",
   "device_class=""temperature",
   "device_id=None",
   "disabled_by=None",
   "entity_category=None",
   "icon=None",
   "id=""ab12532e0a3e6f14541e131efb79c1ce",
   "name=None",
   "options="{
      
   },
   "original_device_class=""temperature",
   "original_icon=None",
   "original_name=""Outside Temperature",
   supported_features=0,
   "unit_of_measurement=""°C"
}
```
It does have device class temperature and is in the registry but isn't part of `result`.

Reply:
I also get an empty result to begin with

Reply:
filtering works for me

```python
    result = [
        entity.entity_id
        for entity in filter(
            lambda entity: entity.device_class == device_class
            and entity.domain == Platform.SENSOR,
            _er.entities.values(),
        )
    ]
```

Reply:
but helpers are still missing.